stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/CustomChannelListActivity.kt

```diff
@@ -0,0 +1,29 @@
+package io.getstream.chat.ui.sample.feature.chat
+
+import android.content.Context
+import android.content.Intent
+import android.widget.Toast
+import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.ui.channel.list.ChannelListActivity
+import io.getstream.chat.android.ui.channel.list.ChannelListFragment
+
+class CustomChannelListActivity : ChannelListActivity(), ChannelListFragment.ChannelClickListener {
+
+    override fun createChannelListFragment(): ChannelListFragment {
+        return ChannelListFragment.newInstance {
+            setFragment(CustomChannelListFragment())
+            showSearch(false)
+            showHeader(true)
+        }
+    }
+
+    override fun onChannelClick(channel: Channel) {
+        Toast.makeText(this, "Channel clicked", Toast.LENGTH_SHORT).show()
```

Comment:
Click listener interface (for example, `ChannelListFragment.ChannelClickListener`) can be implemented by parent Fragment or Activity